File: data/readable_format/fortran/file750.f90
MODULE mo_column
  IMPLICIT NONE
CONTAINS
  SUBROUTINE compute(nz, b, q_subarray, t_subarray)
    INTEGER, INTENT(IN) :: nz, b
    REAL, INTENT(INOUT) :: q_subarray(nz), t_subarray(b)
    INTEGER :: i
    ! Increment each element by 1.0
    DO i = 1, nz
      q_subarray(i) = q_subarray(i) + 1.0
    END DO
    DO i = 1, b
      t_subarray(i) = t_subarray(i) + 1.0
    END DO
  END SUBROUTINE compute
END MODULE mo_column

PROGRAM test_abstraction21
  USE mo_column, ONLY: compute
  REAL, DIMENSION(20,60) :: q = 0.0
  REAL, DIMENSION(60,20) :: t = 0.0
  INTEGER :: nproma, nz
  INTEGER :: p

  nproma = 20
  nz = 60

  DO p = 1, nproma
    CALL compute(nz, nproma, q(p,:), t(:,p))
  END DO

  PRINT*, "Sum of q:", SUM(q)
  PRINT*, "Sum of t:", SUM(t)
END PROGRAM test_abstraction21

File: data/readable_format/cpp/file750.cpp
#include <vector>
#include <numeric>
#include <iostream>

// Define the compute function.
void compute(int nz, int b, float* q_subarray, float* t_subarray) {
    for (int i = 0; i < nz; ++i) {
        q_subarray[i] += 1.0f;
    }
    for (int i = 0; i < b; ++i) {
        t_subarray[i] += 1.0f;
    }
}

int main() {
    const int nproma = 20; // Corresponds to columns in q, rows in t
    const int nz = 60;     // Corresponds to rows in q, columns in t

    std::vector<float> q(nproma * nz, 0.0f);
    std::vector<float> t(nz * nproma, 0.0f);

    for (int p = 0; p < nproma; ++p) {
        compute(nz, nproma, &q[p * nz], &t[p * nproma]);
    }

    float sum_q = std::accumulate(q.begin(), q.end(), 0.0f);
    float sum_t = std::accumulate(t.begin(), t.end(), 0.0f);

    std::cout << "Sum of q: " << sum_q << std::endl;
    std::cout << "Sum of t: " << sum_t << std::endl;

    return 0;
}